core/kazoo_documents/src/kzd_devices.erl

```diff
@@ -650,6 +651,22 @@ provision_id(Doc, Default) ->
 set_provision_id(Doc, ProvisionId) ->
     kz_json:set_value([<<"provision">>, <<"id">>], ProvisionId, Doc).
 
+-spec push(doc()) -> kz_json:api_object().
+push(Doc) ->
```

Comment:
generally speaking, "public" fields like this (meaning no `pvt_`) should be defined in the associated schema. Even if the "push" object sub-properties aren't defined at least the key itself should be.